### PR TITLE
conf-postgresql - update depexts package name for macos + homebrew (#23912)

### LINKED ISSUE #23912: conf-postgres depext needs to be updated for homebrew
- related: https://github.com/ocaml/opam-repository/pull/23909 ([detailed in this comment](https://github.com/ocaml/opam-repository/pull/23909#issuecomment-1586182484))

Currently the package `conf-postgres` is specifying `postgresql` as the homebrew package to install on macOS (when using homebrew) https://github.com/ocaml/opam-repository/blob/f016dc51d87d9323b994ea1d40c09a70187d0b12/packages/conf-postgresql/conf-postgresql.1/opam#L27

But recently with homebrew, one has to explicitly specify the major version of postgres to install, e.g. `brew install postgresql@14` or `brew install postgresql@15` and `brew install postgresql` will fail.

### PR DESCRIPTION
**Pull request by haochenx**
- this should fix #23912.

### CODE REVIEW

**Comment by haochenx** [comment]
To "link" keg-only formulas, one can run `brew link postgresql@15`.

But I have no idea how to achieve that in the opam file.
@avsm @mseri  @kit-ty-kate could you provide some advice?

**Review by kit-ty-kate — COMMENTED** [review]
Sorry for the delay.

The behaviour by homebrew is a bit annoying to map in opam but we could do a couple of things:
* Mirror what we did for `conf-libssl` (https://github.com/ocaml/opam-repository/blob/9faf3dbf816f376733b73f3ed8832c4213db4a02/packages/conf-libssl/conf-libssl.4/files/homebrew.sh) and copy the binaries and libraries in the opam switch. However i feel a bit icky doing this for this package (it's fine for libssl as it is a  central package and not much needs to be done anyway) because it's fairly brittle and not the best to maintain.
* Ignore the failures in CI as there isn't that many packages that rely on postgresql in opam and display a `post-message` inviting the user to call `brew link` manually

We could also try to bake this behaviour in opam as a more long term solution. Could you open a ticket in the opam bugtracker to at least get the conversation started? https://github.com/ocaml/opam/issues

For now i think we could go with the second solution with a message looking like that? What do you think?

**Comment by kit-ty-kate** [comment]
Thanks!

File: packages/conf-postgresql/conf-postgresql.1/opam [hunk 1]
```diff
@@ -24,7 +24,7 @@ depexts: [
   ["postgresql14-dev"] {os-distribution = "alpine" & os-version >= "3.15"}
   ["postgresql"] {os = "win32" & os-distribution = "cygwinports"}
   ["postgresql-libs"] {os-family = "arch"}
-  ["postgresql"] {os = "macos" & os-distribution = "homebrew"}
+  ["postgresql@15"] {os = "macos" & os-distribution = "homebrew"}
   ["postgresql96"] {os = "macos" & os-distribution = "macports"}
   ["postgresql"] {os-distribution = "nixos"}
 ]
```
[the remark this answers is not in the document]
Comment (haochenx) [reply packages/conf-postgresql/conf-postgresql.1/opam:30]: thanks!

**Comment by haochenx** [comment]
Many thanks @kit-ty-kate! I have created the issue in ocaml/opam to kick off the discussion as advised.

**Pull request merged**